redbot/cogs/trivia/trivia.py

```diff
@@ -671,9 +672,39 @@ async def _save_trivia_list(
                 return
 
         buffer = io.BytesIO(await attachment.read())
-        yaml.safe_load(buffer)
-        buffer.seek(0)
+        trivia_dict = yaml.safe_load(buffer)
+
+        schema = Schema(
+            {
+                Optional("AUTHOR"): str,
+                Optional("CONFIG"): {
+                    Optional("max_score"): int,
+                    Optional("timeout"): Or(int, float),
+                    Optional("delay"): Or(int, float),
+                    Optional("bot_plays"): bool,
+                    Optional("reveal_answer"): bool,
+                    Optional("payout_multiplier"): Or(int, float),
+                },
+                str: [
+                    str,
+                    int,
+                    bool,
+                    float,
+                ],
+            }
+        )
+        try:
+            schema.validate(trivia_dict)
```

Comment:
This could be put in a separate function which would make it possible to use directly from the existing tests rather than duplicating the schema.